tools/shared/src/main/scala/org/scalajs/core/tools/linker/frontend/optimizer/OptimizerCore.scala

```diff
@@ -3766,7 +3964,7 @@ private[optimizer] object OptimizerCore {
   }
 
   private object PreTransTree {
-    def apply(tree: Tree): PreTransTree = {
+    def make(tree: Tree): PreTransTree = {
```

Comment:
This one was hard. Basically, I wanted `apply` to do *the right thing* for most cases. Basically, if I want to create a `PreTransform` from a `Tree`, I call `PreTransTree(tree)`, and it is smart (and sometimes it doesn't return a `PreTransTree` but something else). `make` is needed for the few cases where we do *not* want it to be smart, which are basically the rare cases where you really want a `PreTransTree` no matter what. And in fact there is only one such place: in `resolveLocalDef`, where you were specifically asked not to be smart anymore, and instead force things to return a `PreTransGenTree`.

An alternative, I guess, would be to give back the meaning of the current `make` to `apply`, and give the meaning of `apply` to a `.toPreTransform` decorator method for `Tree`. That would also be more consistent with `PreTransLocalDef.toPreTransform`.